horovod/mxnet/__init__.py

```diff
@@ -99,11 +99,16 @@ def __init__(self, params, optimizer, optimizer_params=None):
         # average in allreduce, has better performance. 
         self._scale /= size()
 
+        self._keep_param_order = isinstance(params, mx.gluon.parameter.ParameterDict)
```

Comment:
Thanks for adding the improvement. For some rare case, users define the parameter in nondeterministic orders. That would lead to different order of insertion to ParameterDict. I'm not sure how to easily workaround that, unless all workers exchange the parameter order and name at initialization. What do you think?

Reply:
Actually, I think there might be a better solution. Right now, the requests to Horovod are being named with `str(i)` which is what is causing the issue with the ordering of the loop. If we instead name the requests with the actual parameter names instead, the order they are submitted will not matter. In the case of inconsistently ordered parameters, the priorities will be set differently across workers, but that would also happen in the native `trainer.py` with kvstore. Does this alternate solution sound alright to you?

Reply:
Oh you're right. I like this solution much better. 
BTW - do you find the priority helpful in scheduling horovod allreduce? Any perf benchmark u can share?

Reply:
Ok great! I will update the PR.

I do not currently have any extensive perf benchmarks for this.